src/apps/wallet/sign_tx/scripts.py

```diff
@@ -42,6 +42,16 @@ def output_script_p2sh(scripthash: bytes) -> bytearray:
     return s
 
 
+def script_replay_protection_bip115(block_hash: bytes, block_height: bytes) -> bytearray:
+    block_height_size = len(block_height)  # SIZE in bytes block heigh (should be 3)
+    s = bytearray(38)
+    s[0] = 0x20  # 32 bytes for block hash
+    s[1:33] = block_hash  # block hash
+    s[33] = 0x03  # 3 bytes for block height
+    s[34:37] = (block_height if (block_height_size == 3) else block_height[:3 - block_height_size])  # MUST BE ONLY 3 BYTES FOR BLOCKHEIGHT
```

Comment:
Is this behaviour somewhere described? This does not seem very clever

Reply:
Pavol, thank you for your comment.

Replay protection needs a valid block height and a valid block hash, otherwise the transaction will not be spendable. If we rely on the correctness of the passed block height size, it is also possible to remove the check.
Regarding the following commit: (9f90ca2#diff-711e9d4abbb3c6798636eb0324377c3eL279) Please consider that Replay protection has been introduced in ZenCash after a certain block height, therefore not every transaction will have it specified in the script. For this reason, if someone tries to spend an old transaction, it will not be needed to consider replay protection block hash and block height to reconstruct the script to be signed. In this case it will be needed to keep the check.

For example, please take a look at the following transaction: https://explorer.zensystem.io/insight-api-zen/tx/5dbcdfd462a6fef34e97ecc0d137bd172ad0be213ba985c1327af3ae3277ab95.

Reply:
I get it. I fixed this behaviour in 6d5a8645fd1d1d4768d8b74b90125fd90b97bf79 - when None is passed to the script_bip115 function, an empty script is returned which is the behaviour you had originally.

Can you describe the magic about the `block_height` - where is the fact that it's always supposed to be 3 bytes documented?

Reply:
Pavol, it was a way to prevent to write a wrong block height if it was passed with more than 3 bytes. as explained before, if we rely on correctness of such block height size, we can remove the check.

Reply:
I think we should pass this value as `uint32` and convert to varint bytes, if you agree I'll do the change.